src/applications/personalization/dashboard/helpers.jsx

```diff
@@ -195,6 +196,31 @@ export function isSIPEnabledForm(savedForm) {
   return true;
 }
 
+// Callback to use with Array.sort that expects two properly formatted form
+// objects (that have `metadata.expiresAt` properties). Used to sort form
+// objects, placing the form that expires sooner before the form that expires
+// later
+export function sipFormSorter(formA, formB) {
+  // simple helper to make sure the arg is an object with a metadata.expiresAt
+  // prop
+  function isValidForm(arg) {
+    if (!isPlainObject(arg)) {
+      throw new TypeError(`${arg} is not a plain object`);
+    }
+    if (
+      !arg.metadata ||
+      !arg.metadata.expiresAt ||
+      typeof arg.metadata.expiresAt !== 'number'
+    ) {
+      throw new TypeError(`'metadata.expiresAt' is not set on ${arg}`);
+    }
+    return true;
+  }
+
+  [formA, formB].forEach(isValidForm);
```

Comment:
I'm wide open to suggestions for different/better ways to validate the args that are passed in. This totally works, but I'm not crazy about it

Reply:
You could add this as another step for creating the `verifiedSavedForms`:
``` 
const verifiedSavedForms = savedForms
	    .filter(isSIPEnabledForm)
            .forEach(throwIfInvalidForm) 
	    .sort(sipFormSorter);
```

Otherwise the sort is could end up validating the same form multiple times.

Reply:
I think I like that...